src/main/java/seedu/address/timetable/TimetableRetriever.java

```diff
@@ -0,0 +1,123 @@
+
+        for (int moduleIter = 0; moduleIter < moduleCodeArray.length; moduleIter++) {
+            String urlString = getUrlString(moduleCodeArray[moduleIter]);
+            URL url = new URL(urlString);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.connect();
+            int responseCode = connection.getResponseCode();
+            if (responseCode != 200) {
+                throw new RuntimeException("HttpResponseCode: " + responseCode);
+            }
+
+            String inline = "";
+            Scanner sc = new Scanner(url.openStream());
+            while (sc.hasNext()) {
+                inline += sc.nextLine();
+            }
+
+            JSONParser jsonParser = new JSONParser();
+            JSONObject moduleData = (JSONObject) jsonParser.parse(inline);
+            JSONArray semesterData = (JSONArray) moduleData.get("semesterData");
+            JSONObject bothSemesterTimetableData = (JSONObject) semesterData.get(semester - 1);
+            JSONArray semesterSpecificTimetableData = (JSONArray) bothSemesterTimetableData.get("timetable");
+
+            String[] specificModuleLessonArray = getSpecificModuleLessonArray(moduleLessonArray, moduleIter);
+
+            for (int lessonIter = 0; lessonIter < specificModuleLessonArray.length; lessonIter++) {
+                if (specificModuleLessonArray[0].equals("")) {
+                    System.out.println(moduleCodeArray[moduleIter] + ": no lessons");
+                } else {
+                    String lessonInfo = specificModuleLessonArray[lessonIter];
+                    String lessonType = getLessonType(lessonInfo);
+                    String lessonNum = getLessonNum(lessonInfo);
+                    for (int dataIter = 0; dataIter < semesterSpecificTimetableData.size(); dataIter++) {
+                        JSONObject currentData = (JSONObject) semesterSpecificTimetableData.get(dataIter);
+                        String currentLessonType = (String) currentData.get("lessonType");
+                        String currentLessonNum = (String) currentData.get("classNo");
+                        if (currentLessonType.equals(lessonType) && currentLessonNum.equals(lessonNum)) {
+                            String day = (String) currentData.get("day");
+                            String startTime = DUMMY_DATE + (String) currentData.get("startTime");
+                            String endTime = DUMMY_DATE + (String) currentData.get("endTime");
+                            String module = moduleCodeArray[moduleIter];
+                            String name = module + currentLessonType;
+                            lessons.add(new Lesson(new Name(name), new Deadline(startTime), new Deadline(endTime),
+                                    new ModuleCode(module)));
+                        }
+                    }
+                }
```

Comment:
According to the CS2103T coding standards/conventions, we should try to avoid deeply nested methods. Maybe you could consider removing the else block and extracting the "happy path" to prevent the method from becoming too deeply nested. Another suggestion is adding an empty line in between lines of code to improve readability. You could consider applying the SLAP framework as well :)

Reply:
Amended!